adoc/admin-disaster-recovery-velero.adoc

```diff
@@ -0,0 +1,707 @@
+
+** Manual
++
+[source,bash]
+----
+velero backup create <BACKUP_NAME>
+----
+
+** Schedule
++
+The schedule templatein in cron notation, using UTC time:
++
+|=======
+| Character Position | Character Period | Acceptable Values
+| 1 | Minute | 0-59,*
+| 2 | Hour | 0-23,*
+| 3 | Day of Month | 1-31,*
+| 4 | Month | 1-12,*
+| 5 | Day of Week | 0-7,*
+|=======
++
+The schedule can also be expressed using "@every <duration>" syntax. The duration
+can be specified using a combination of seconds (s), minutes (m), and hours (h), for
+example: "@every 2h30m".
++
+[source,bash]
+----
+# Create schedule template
+# Create a backup every 6 hours
+velero schedule create <SCHEDULE_NAME> --schedule="0 */6 * * *"
+
+# Create a backup every 6 hours with the @every notation
+velero schedule create <SCHEDULE_NAME> --schedule="@every 6h"
+
+# Create a daily backup of the web namespace
+velero schedule create <SCHEDULE_NAME> --schedule="@every 24h" --include-namespaces web
+
+# Create a weekly backup, each living for 90 days (2160 hours)
+velero schedule create <SCHEDULE_NAME> --schedule="@every 168h" --ttl 2160h0m0s
+----
+
+** Optional Flags
+
+*** Granularity
+
+**** Cluster
++
+Without pass extra flags to `velero backup create`, Velero will backup whole {kube} cluster.
+
+**** Namespace
++
+Pass flag `--include-namespaces` or `--exclude-namespaces` to specifies which namespaces to include/exclude when backing up. For example:
++
+[source,bash]
+----
+# Create a backup including the nginx and default namespaces
+velero backup create backup-1 --include-namespaces nginx,default
+
+# Create a backup excluding the kube-system and default namespaces
+velero backup create backup-1 --exclude-namespaces kube-system,default
+----
+
+**** Resources
++
+Pass flag `--include-resources` or `--exclude-resources` to specifies which resources to include/exclude when backing up. For example:
++
+[source,bash]
+----
+# Create a backup including storageclass resource only
+velero backup create backup-1 --include-resources storageclasses
+----
++
+[TIP]
+Use `kubectl api-resources` to lists all API resources on the server.
+
+**** Label Selector
++
+Pass `--selector` to only back up resources matching the label selector.
++
+[source,bash]
+----
+# Create a backup for the elasticsearch cluster only
+velero backup create backup-1 --selector app=elasticsearch-master
+----
+
+*** Location
++
+Pass `--storage-location` to specify where the backup stores to. For example, if we have HA object storage server called master and slave respectively.
++
+[source,bash]
+----
+# Create a backup to master storage server
+velero backup create backup2master --storage-location master
+
+# Create a backup to slave storage server
+velero backup create backup2slave --storage-location slave
+----
+
+*** Garbage collection
++
+Pass `--ttl` to determine how long keeps the backup, after that, the backup will be garbage collected. The default backup existed time is 720 hours (30 days).
```

Comment:
@c3y1huang this is kind of file retention.  (This is just a comment :) )

Reply:
For etcd backup cronjob can control retention at job level but the not the actual etcd snapshot stored. I will add in doc as HINT for user. Ref: https://github.com/SUSE/doc-caasp/pull/682.